crates/libcgroups/src/common.rs

```diff
@@ -92,39 +161,103 @@ pub struct ControllerOpt<'a> {
     pub freezer_state: Option<FreezerState>,
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum WrappedIoError {
+    #[error("failed to open {path}: {err}")]
+    Open { err: std::io::Error, path: PathBuf },
+    #[error("failed to write {data} to {path}: {err}")]
+    Write {
+        err: std::io::Error,
+        path: PathBuf,
+        data: String,
+    },
+    #[error("failed to read {path}: {err}")]
+    Read { err: std::io::Error, path: PathBuf },
+    #[error("failed to create dir {path}: {err}")]
+    CreateDir { err: std::io::Error, path: PathBuf },
```

Comment:
I love this WrappedIoError idea ❤️ 
If the dirs cannot be created in the libcgroup crates, does this result in the failure to create a new cgroup?
I believe it is more user-friendly to display a message such as "failed to create the cgroup" or something similar. WDYT?

Reply:
That's what errors higher up in the chain are for - they can be added to the higher-level error adts with more user-friendly messages